tools/qontract_cli.py

```diff
@@ -1103,83 +1103,97 @@ def app_interface_merge_queue(ctx):
+            if "stale" in labels:
+                continue
+            if SAAS_FILE_UPDATE in labels:
+                continue
+            if SELF_SERVICEABLE in labels:
+                continue
 
-        is_assigned_by_app_sre = gl.is_assigned_by_team(mr, app_sre_team_members)
-        if is_assigned_by_app_sre:
-            continue
+            pipelines = mr.pipelines()
+            if not pipelines:
+                continue
+            running_pipelines = [p for p in pipelines if p["status"] == "running"]
+            if running_pipelines:
+                continue
+            last_pipeline_result = pipelines[0]["status"]
+            if last_pipeline_result != "success":
+                continue
 
-        is_last_action_by_app_sre = gl.is_last_action_by_team(
-            mr, app_sre_team_members, glhk.HOLD_LABELS
-        )
+            author = mr.author["username"]
+            app_sre_team_members = [u.username for u in gl.get_app_sre_group_users()]
+            if author in app_sre_team_members:
+                continue
 
-        if is_last_action_by_app_sre:
-            last_comment = gl.last_comment(mr, exclude_bot=True)
-            # skip only if the last comment isn't a trigger phrase
-            if last_comment and not re.fullmatch(
-                trigger_phrases_regex, last_comment["body"]
-            ):
+            is_assigned_by_app_sre = gl.is_assigned_by_team(mr, app_sre_team_members)
+            if is_assigned_by_app_sre:
                 continue
 
-        item = {
-            "id": f"[{mr.iid}]({mr.web_url})",
-            "title": mr.title,
-            "onboarding": "onboarding" in labels,
-            "updated_at": mr.updated_at,
-            "author": author,
-            "labels": ", ".join(labels),
-        }
-        queue_data.append(item)
+            is_last_action_by_app_sre = gl.is_last_action_by_team(
+                mr, app_sre_team_members, glhk.HOLD_LABELS
+            )
+
+            if is_last_action_by_app_sre:
+                last_comment = gl.last_comment(mr, exclude_bot=True)
+                # skip only if the last comment isn't a trigger phrase
+                if (
+                    last_comment
+                    and trigger_phrases_regex
+                    and not re.fullmatch(trigger_phrases_regex, last_comment["body"])
+                ):
+                    continue
+
+            item = {
+                "id": f"[{mr.iid}]({mr.web_url})",
+                "repo": repo,
+                "title": mr.title,
+                "onboarding": "onboarding" in labels,
+                "updated_at": mr.updated_at,
+                "author": author,
+                "labels": ", ".join(labels),
+            }
+            queue_data.append(item)
+        return queue_data
+
+    queue_data = []
+
+    review_repos = queries.get_review_repos()
+    if review_repos:
+        for repo in review_repos:
+            queue_data.extend(get_mrs(repo["name"], repo["url"]))
```

Comment:
Just a nit. `queries.get_review_repos()` always returns a list. Therefore this can be written as: 

```python
for repo in queries.get_review_repos():
    queue_data.extend(get_mrs(repo["name"], repo["url"]))
```